cmd/ci-secret-bootstrap/main.go

```diff
@@ -242,12 +248,20 @@ type secretConfig struct {
 	To   []secretContext             `json:"to"`
 }
 
-func constructSecrets(config []secretConfig, bwClient bitwarden.Client) (map[string][]*coreapi.Secret, error) {
+func constructSecrets(ctx context.Context, config []secretConfig, bwClient bitwarden.Client, maxConcurrency int) (map[string][]*coreapi.Secret, error) {
+	sem := semaphore.NewWeighted(int64(maxConcurrency))
 	secretsMap := map[string][]*coreapi.Secret{}
 	secretsMapLock := &sync.Mutex{}
 
+	errChan := make(chan error)
 	var errs []error
-	errLock := &sync.Mutex{}
+	errReadingDone := make(chan struct{})
```

Comment:
Accumulate the errors after the works have finished, and you don't need this.

Reply:
My first version was like that (having no `errReadingDone`). But I had hard time to make the tests stable.
Until I found similar code:

https://github.com/openshift/ci-tools/blob/53a01129e7320d7903abc73423b8c95d5cf6a3aa/cmd/sanitize-prow-jobs/main.go#L114-L120

Then all tests are green

```
go test -count=100 -run TestConstructSecrets -v ./cmd/ci-secret-bootstrap/...
```

Reply:
?

Reply:
Updated.